### PR TITLE
Unified Query Params for compatible with different storages 

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

As titled

### CODE REVIEW

File: research/gaia/gremlin/gremlin_core/src/structure/graph.rs [hunk 1]
```diff
@@ -33,52 +40,78 @@ impl<E: Element + Send + Sync> Default for QueryParams<E> {
     }
 }
 
+impl<E: Element + Send + Sync> FromPb<Option<pb::QueryParams>> for QueryParams<E> {
+    fn from_pb(query_params_pb: Option<pb::QueryParams>) -> Result<Self, ParseError>
+    where
+        Self: Sized,
+    {
+        let mut query_params = QueryParams::default();
+        if let Some(query_params_pb) = query_params_pb {
```
Comment (longbinlai) [inline research/gaia/gremlin/gremlin_core/src/structure/graph.rs:49]: I am suggesting the following codes:
```
Implement the follow functions<where xxx is labels, limit, required_properties, predicates, extra_params> in QueryParams: 
fn with_xxx(mut self, Option<xxx>) -> Self
```
Then in from_pb, you can do:
```
let mut query_params = QueryParams::default();
if let Some(query_params_pb) = query_params_pb {
      query_params = query_params.with_labels_from_pb(labels)
        .with_limit(limit)
        .with_required_properties(xx)
        .with_predicates(xx)
        .with_extra_params(xx);
}
```

Comment (BingqingLyu) [reply research/gaia/gremlin/gremlin_core/src/structure/graph.rs]: Done

File: research/gaia/gremlin/gremlin_core/src/structure/graph.rs [hunk 2]
```diff
@@ -45,73 +45,85 @@ impl<E: Element + Send + Sync> FromPb<Option<pb::QueryParams>> for QueryParams<E
...
-                query_params.set_extra_params(extra_params)
-            }
-        }
-        Ok(query_params)
+        query_params_pb.map_or(Ok(QueryParams::default()), |query_params_pb| {
+            QueryParams::default()
+                .with_labels(query_params_pb.labels)?
+                .with_filter(query_params_pb.predicates)?
+                .with_limit(query_params_pb.limit)?
+                .with_props(query_params_pb.required_properties)?
```
Comment (longbinlai) [inline research/gaia/gremlin/gremlin_core/src/structure/graph.rs:53]: I'd like the name of `with_required_properties`.

Comment (BingqingLyu) [reply research/gaia/gremlin/gremlin_core/src/structure/graph.rs]: Done

File: research/gaia/gremlin/compiler/gremlin-server-plugin/src/main/java/com/alibaba/graphscope/gaia/plan/translator/builder/PlanConfig.java [hunk 3]
```diff
@@ -19,4 +19,5 @@
     public static String QUERY_ID = "query_id";
     public static String TAG_ID_MAKER = "tag_id_maker";
     public static String QUERY_CONFIG = "query_config";
+    public static String SNAPSHOT_ID = "snapshot_id";
```
Comment (longbinlai) [inline research/gaia/gremlin/compiler/gremlin-server-plugin/src/main/java/com/alibaba/graphscope/gaia/plan/translator/builder/PlanConfig.java:22]: Add a TODO() for aligning with the snapshot id in graph adaptor

**Review by longbinlai — APPROVED** [review]

**Pull request merged**